test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -4021,5 +4021,236 @@ public void PackCommand_Deterministic_MultiplePackInvocations_CreateIdenticalPac
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\media\\icon.jpg", 10)
+                .WithFile("test\\media\\example.txt", 10)
+                .WithFile("test\\media\\readme.txt", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("media\\icon.jpg")
+                .WithItem("None", "media\\*", "media");
+
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_FolderNested_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\folder\\nested\\icon.jpg", 10)
+                .WithFile("test\\folder\\nested\\sample.txt", 10)
+                .WithFile("test\\folder\\nested\\media\\readme.txt", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("media\\nested\\icon.jpg")
+                .WithItem("None", "folder\\**", "media");
+                
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_FolderNestedFlatPack_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();   
+
+            testDirBuilder
+                .WithFile("test\\folder\\nested\\icon.jpg", 10)
+                .WithFile("test\\folder\\nested\\sample.txt", 10)
+                .WithFile("test\\folder\\nested\\media\\readme.txt", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("nested\\icon.jpg")
+                .WithItem("None", "folder\\**", string.Empty);
+
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_PackageIconUrl_Suceeds()
```

Comment:
To reduce duplication this test can both test that the whole operation is successful and verify the warning.

Reply:
Will combine this test with the happy path.